marlin/src/marlin/circuit_verifying_key.rs

```diff
@@ -53,6 +60,26 @@ impl<F: PrimeField, CF: PrimeField, PC: PolynomialCommitment<F, CF>> ToBytes for
     }
 }
 
+impl<F: PrimeField, CF: PrimeField, PC: PolynomialCommitment<F, CF>> ToMinimalBitRepresentation
+    for CircuitVerifyingKey<F, CF, PC>
+{
+    fn to_minimal_bit_representation(&self) -> Vec<bool> {
+        let domain_h = EvaluationDomain::<F>::new(self.circuit_info.num_constraints)
+            .ok_or(SynthesisError::PolynomialDegreeTooLarge)
+            .unwrap();
+        let domain_k = EvaluationDomain::<F>::new(self.circuit_info.num_non_zero)
+            .ok_or(SynthesisError::PolynomialDegreeTooLarge)
+            .unwrap();
+
+        let domain_h_size_bits = CF::from(domain_h.size() as u128).to_bits_le();
+        let domain_k_size_bits = CF::from(domain_k.size() as u128).to_bits_le();
```

Comment:
We don't need to serialize things as u128; u64 should be more than enough.

Reply:
Updated. Now only the 64 bits are being included in the minimal bit representations, and the remaining bits are enforced to be smaller than u64::MAX, both in the outside and in the constraint world.